azure-communication-ui/azure-communication-ui-demo-app/src/androidTest/java/com/azure/android/communication/ui/callingcompositedemoapp/permissions/BaseUiPermissionTest.kt

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.callingcompositedemoapp.permissions
+
+import android.os.Build
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.rule.GrantPermissionRule
+import com.azure.android.communication.ui.callingcompositedemoapp.CallLauncherActivity
+import com.azure.android.communication.ui.callingcompositedemoapp.append
+import com.microsoft.appcenter.espresso.Factory
+import com.microsoft.appcenter.espresso.ReportHelper
+import leakcanary.DetectLeaksAfterTestSuccess
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+
+open class BaseUiPermissionTest {
+    @get:Rule
+    var reportHelper: ReportHelper = Factory.getReportHelper()
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityScenarioRule(CallLauncherActivity::class.java)
+
+    @Rule
+    @JvmField
+    var grantPermissionRule: GrantPermissionRule
+
+    @get:Rule
+    val rule = DetectLeaksAfterTestSuccess()
+
+    private val basePermissionList = arrayOf(
+        "android.permission.ACCESS_NETWORK_STATE",
+        "android.permission.WAKE_LOCK",
+        "android.permission.MODIFY_AUDIO_SETTINGS",
+    )
+
+    init {
+        val permissionList = if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
```

Comment:
for API 31 devices we need the `BLUETOOTH_CONNECT` permissions.  Can you move this into a function (kind of similar to `PermissionsManager.getPermissionsList()` but inside a companion object) so that it can be re-used by `BaseUiTest()` later